module/icp/algs/impl/impl.c

```diff
@@ -0,0 +1,175 @@
+
+#include <sys/simd.h>
+#include <impl/impl.h>
+
+#define	ALG_IMPL_READ(i) (*(volatile uint32_t *) &(i))
```

Comment:
Is there a common library function that does the same thing?

Reply:
I do not pretend to be an expert, but perhaps [atomic_load_32()](https://github.com/openzfs/zfs/blob/master/include/os/linux/spl/sys/atomic.h#L51)?

Reply:
Done.